File: 1002-find-common-characters/1002-find-common-characters.cpp
class Solution {
public:
    vector<string> commonChars(vector<string>& words) {
        vector<int>mainCounter(26, 0);
        string start = words[0];
        for(char i : start){
            ++mainCounter[i - 'a'];
        }
        for(int i = 1; i < words.size(); i++){
            vector<int>temp(26, 0);
            string word = words[i];
            
            for(char j : word){
                ++temp[j - 'a'];
            }
            
            for(int k = 0; k < 26; k++){
                mainCounter[k] = min(mainCounter[k], temp[k]);
            }
        }
        
        vector<string>ans;
        for(int i = 0; i < 26; i++){
            while(mainCounter[i]--){
                string t = "";
                char d = i + 'a';
                t.push_back(d);
                ans.push_back(t);
            }
        }
        return ans;
    }
};

File: 1002-find-common-characters/1002-find-common-characters.java
class Solution {
    public List<String> commonChars(String[] words) {
        
        int main[] = new int[26];
        
        String s = words[0];
        
        for(int i = 0; i < s.length(); i++)
            ++main[s.charAt(i) - 'a'];
        
        for(int i = 1; i < words.length; ++i){
            
            s = words[i];
            
            int temp[] = new int[26];
            
            for(int j = 0; j < s.length(); j++)
                ++temp[s.charAt(j) - 'a'];
            
            for(int k = 0; k < 26; k++)
                main[k] = Math.min(main[k], temp[k]);
                          
        }
        
        List<String>res = new ArrayList<>();
        
        for(int i = 0; i < 26; i++){
            
            while(main[i] != 0){
                
                String ch = String.valueOf((char) (i + 'a'));
                
                res.add(ch);
                
                --main[i];
                
            }
            
        }
        
        return res;
        
    }
}